File: src/mod10_design_pattern/lesson3_abstract_factory/java/example_with_abstract_class/Owl.java
package mod10_design_pattern.lesson3_abstract_factory.java.example_with_abstract_class;

public class Owl extends AbstractMagicalCreature {
    @Override
    public void makeSound() {
        System.out.println("Uuuuu");
    }

    @Override
    public String getDescription() {
        return "The owl is wise creature that serves as a messanger.";
    }
}

File: src/mod10_design_pattern/lesson3_abstract_factory/kotlin/Owl.kt
package mod10_design_pattern.lesson3_abstract_factory.kotlin

class Owl:AbstractMagicalCreature() {
    override fun makeSound() {
        println("Uuuuu")
    }

    override fun getDescription(): String {
        return "The owl is a wise creature that serves as a messenger."
    }
}